File: matiasmasso/Mat.NET/DT2/DTODefaultImage.vb
Public Class DTODefaultImage
    Property Id As Defaults.ImgTypes
    Property Image As Image

    Shared Function Factory(oId As Defaults.ImgTypes, Optional oImage As Image = Nothing)
        Dim retval As New DTODefaultImage
        With retval
            .Id = oId
            .Image = oImage
        End With
        Return retval
    End Function
End Class

File: matiasmasso/Mat.NET/DTO/DTODefaultImage.cs

namespace DTO
{
    public class DTODefaultImage
    {
        public Defaults.ImgTypes Id { get; set; }
        public byte[] Image { get; set; }

        public static DTODefaultImage Factory(Defaults.ImgTypes oId, byte[] oImage = null/* TODO Change to default(_) if this is not a reference type */)
        {
            DTODefaultImage retval = new DTODefaultImage();
            {
                var withBlock = retval;
                withBlock.Id = oId;
                withBlock.Image = oImage;
            }
            return retval;
        }
    }
}
